code/modules/mob/living/carbon/damage_procs.dm

```diff
@@ -101,6 +101,7 @@
 		take_overall_damage(0, 0, amount, updating_health)
 	else
 		heal_overall_damage(0, 0, abs(amount), FALSE, FALSE, updating_health)
+	SEND_SIGNAL(src, COMSIG_LIVING_ADJUST_DAMAGE, STAMINA, amount)
```

Comment:
You should be having different signals per damage type and only have the signal on the base proc. You may need to refactor things so they all actually call on the parent.

Also is you make the special list `args` one of the signal arguments it will let you modify the amount inside the receiver proc.